### ISSUE TITLE
SDPA Int8 Quantisation using MTQ

### ISSUE THREAD

**Issue by satya-penamakuri** [NVIDIA/TensorRT-Model-Optimizer#170]
I'm attempting to quantize SDPA similarly to TRT's approach—by adding Q-DQ on the Q, K, V, and softmax output. I've observed that MOQ implements this, whereas MTQ does not. Additionally, I would like to perform QAT after adding quantizers via MTQ. Does MTQ offer kernels to mitigate training slowdowns? This is crucial since SDPA typically features efficient implementations, and the quantization process requires inserting a Q-DQ at an intermediate output.

<img width="543" alt="Image" src="https://github.com/user-attachments/assets/373bff9d-a503-4db2-ab24-2238accb6348" />

**Comment by jingyu-ml** [comment]
@satya-penamakuri What's the target model you're working with?

**Comment by satya-penamakuri** [comment]
it's a transformer model

**Comment by jingyu-ml** [comment]
QAT can't be applied to the projection (softmax output) when using SDPA, but we can apply QAT to the QKV outputs, specifically the inputs to bmm1 and bmm2. because the softmax is rooted in the sdpa kernel code and there's no way we can access to it in torch level during training. 

You can take a look at this example: https://github.com/NVIDIA/TensorRT-Model-Optimizer/issues/149#issuecomment-2705843144 to how to quantize your sdpa, then you can load the quantize ckpt to training directly.
